packages/functions/src/texture-compress.ts

```diff
@@ -228,6 +161,100 @@ export const textureCompress = function (_options: TextureCompressOptions): Tran
+	const logger = document.getLogger();
+	
+	const options = { ...TEXTURE_COMPRESS_DEFAULTS, ..._options } as Required<TextureCompressOptions>;
+	const encoder = options.encoder as typeof sharp | null;
+	const targetFormat = options.targetFormat as Format | undefined;
+
+	if (!encoder) {
+		throw new Error(`${targetFormat}: encoder dependency required — install "sharp".`);
+	}
+
+	const resize = options.resize as vec2 | undefined;
+	const resizeFilter = options.resizeFilter as TextureResizeFilter;
+
+	const textureLabel =
+		texture.getURI() ||
+		texture.getName() ||
+		`${textureIndex + 1}/${document.getRoot().listTextures().length}`;
+	const prefix = `${NAME}(${textureLabel})`;
+
+	const srcFormat = getFormat(texture);
+	const dstFormat = targetFormat || srcFormat;
+	const srcMimeType = texture.getMimeType();
+	const dstMimeType = `image/${dstFormat}`;
+
+	// COMPRESS: Run compression library.
+
+	let encoderOptions: sharp.JpegOptions | sharp.PngOptions | sharp.WebpOptions | sharp.AvifOptions = {};
+
+	switch (dstFormat) {
+		case 'jpeg':
+			encoderOptions = { quality: options.quality } as sharp.JpegOptions;
+			break;
+		case 'png':
+			encoderOptions = {
+				quality: options.quality,
+				effort: remap(options.effort, 100, 10),
+			} as sharp.PngOptions;
+			break;
+		case 'webp':
+			encoderOptions = {
+				quality: options.quality,
+				effort: remap(options.effort, 100, 6),
+				lossless: options.lossless,
+				nearLossless: options.nearLossless,
+			} as sharp.WebpOptions;
+			break;
+		case 'avif':
+			encoderOptions = {
+				quality: options.quality,
+				effort: remap(options.effort, 100, 9),
+				lossless: options.lossless,
+			} as sharp.AvifOptions;
+			break;
+	}
+
+	const srcImage = texture.getImage()!;
+	const instance = encoder(srcImage).toFormat(dstFormat, encoderOptions);
+
+	// Resize.
+	if (resize) {
+		instance.resize(resize[0], resize[1], {
+			fit: 'inside',
+			kernel: resizeFilter,
+			withoutEnlargement: true,
+		});
+	}
+
+	const dstImage = BufferUtils.toView(await instance.toBuffer());
+
+	const srcByteLength = srcImage.byteLength;
+	const dstByteLength = dstImage.byteLength;
+
+	let flag = '';
+
+	if (srcMimeType === dstMimeType && dstByteLength >= srcByteLength) {
+		// Skip if src/dst formats match and dst is larger than the original.
+		flag = ' (SKIPPED)';
+	} else if (srcMimeType === dstMimeType) {
+		// Overwrite if src/dst formats match and dst is smaller than the original.
+		texture.setImage(dstImage);
+	} else {
+		// Overwrite, then update path and MIME type if src/dst formats differ.
+		const srcExtension = ImageUtils.mimeTypeToExtension(srcMimeType);
+		const dstExtension = ImageUtils.mimeTypeToExtension(dstMimeType);
+		const dstURI = texture.getURI().replace(new RegExp(`\\.${srcExtension}$`), `.${dstExtension}`);
+		texture.setImage(dstImage).setMimeType(dstMimeType).setURI(dstURI);
+	}
+
+	logger.debug(`${prefix}: Size = ${formatBytes(srcByteLength)} → ${formatBytes(dstByteLength)}${flag}`);
```

Comment:
This log statement is the only thing requiring a `textureIndex` parameter in the method signature, right? Maybe we should move the logging out of the `compressTexture` function, and do the logging in the transform instead. Being able to call...

```js
compressTexture(texture, {sharp, ...});
```

... would feel cleaner too.

Reply:
Yes I wasn't sure about that but I agree, if logging is needed/desired the caller should be responsible for it in that case